enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AbstractRoleRepository.java

```diff
@@ -44,14 +45,16 @@
     /** Master list of roles */
     protected volatile List<RoleRecord> roles = new ArrayList<>();
 
+    private final Pattern roleNamePattern = Pattern.compile( "^[a-zA-Z0-9_]+$" );
+
     @Override
-    public RoleRecord getRoleByName( String roleName )
+    public synchronized RoleRecord getRoleByName( String roleName )
```

Comment:
rolesByName is a ConcurrentHashMap, so there should be no need for this method to be synchronized.